from main import *
import json

with open('config.json') as file:
    data = json.load(file)

THEME = 'dark' if data["theme"] == 'dark' else 'light'

class ThemeFunctions(Ui_Browser):
    def load_theme(self):
        global THEME

        if THEME == "light":
            self.ui.title_frame.setStyleSheet("QFrame#title_frame {\n"
"    background-color: #898989;\n"
"}\n"
"\n"
"QPushButton {\n"
"    background-color: rgba(0,0,0,0);\n"
"    color: #484848;\n"
"    font: bold;\n"
"    font-size: 15px;\n"
"    font-family: entypo;\n"
"}\n"
"\n"
"QPushButton:hover {\n"
"    color: #E8960C;\n"
"}\n"
"\n"
"QPushButton:pressed {\n"
"    color: #FF8000;\n"
"    padding-left: 5px;\n"
"    padding-top: 5px;\n"
"}")
            self.ui.lineEdit.setStyleSheet("background-color: #8f8f8f;\n"
"border-radius: 5px;\n"
"color: #5c5c5c;\n"
"padding-left: 8px;")
            self.ui.tabWidget.setStyleSheet("QTabWidget::scroller {\n"
"    width: 0px;\n"
"}\n"
"\n"
"QTabWidget::pane {\n"
"    background: #96989f;\n"
"}\n"
"\n"
"QTabWidget::tab-bar:top {\n"
"    top: 1px;\n"
"}\n"
"\n"
"QTabWidget::tab-bar:bottom {\n"
"    bottom: 1px;\n"
"}\n"
"\n"
"QTabWidget::tab-bar:left {\n"
"    right: 1px;\n"
"}\n"
"\n"
"QTabWidget::tab-bar:right {\n"
"    left: 1px;\n"
"}\n"
"\n"
"QTabBar::tab:selected {\n"
"    background: #96989f;\n"
"    color: #000;\n"
"}\n"
"\n"
"QTabBar::tab:!selected {\n"
"    border-right: 1px solid #5d656d;\n"
"    border-left: 1px solid #5d656d;\n"
"    background: #8a8d97;\n"
"    color: #5a6068;\n"
"}\n"
"\n"
"QTabBar::tab:!selected:hover {\n"
"    background: #93959d;\n"
"    color: #5a6068;\n"
"}\n"
"\n"
"QTabBar::tab:top:!selected {\n"
"    margin-top: 1px;\n"
"}\n"
"\n"
"QTabBar::tab:bottom:!selected {\n"
"    margin-bottom: 1px;\n"
"}\n"
"\n"
"QTabBar::tab:top, QTabBar::tab:bottom {\n"
"    min-width: 24ex;\n"
"    margin-right: -1px;\n"
"    padding: 10px;\n"
"}\n"
"\n"
"QTabBar::tab:top:selected {\n"
"    border-bottom-color: none;\n"
"}\n"
"\n"
"QTabBar::tab:bottom:selected {\n"
"    border-top-color: none;\n"
"}\n"
"\n"
"QTabBar::tab:top:last, QTabBar::tab:bottom:last,\n"
"QTabBar::tab:top:only-one, QTabBar::tab:bottom:only-one {\n"
"    margin-right: 0;\n"
"}\n"
"\n"
"QTabBar::tab:left:!selected {\n"
"    margin-right: 3px;\n"
"}\n"
"\n"
"QTabBar::tab:right:!selected {\n"
"    margin-left: 3px;\n"
"}\n"
"\n"
"QTabBar::tab:left, QTabBar::tab:right {\n"
"    min-height: 8ex;\n"
"    margin-bottom: -1px;\n"
"    padding: 10px 5px 10px 5px;\n"
"}\n"
"\n"
"QTabBar::tab:left:selected {\n"
"    border-left-color: none;\n"
"}\n"
"\n"
"QTabBar::tab:right:selected {\n"
"    border-right-color: none;\n"
"}\n"
"\n"
"QTabBar::tab:left:last, QTabBar::tab:right:last,\n"
"QTabBar::tab:left:only-one, QTabBar::tab:right:only-one {\n"
"    margin-bottom: 0;\n"
"}\n"
"\n"
"QTabBar::close-button {\n"
"    background: rgba(0,0,0,0);\n"
"    image: url(:/img/img/cross.png);\n"
"    subcontrol-position: left;\n"
"    border-radius: 7px;\n"
"    padding: 2px;\n"
" }\n"
"\n"
"QTabBar::close-button:hover {\n"
"    background: #a6a6a9;\n"
"cursor: pointing-hand\n"
"}")
            self.ui.bottom_frame.setStyleSheet("background-color: #969696;")

            self.ui.content_frame.setStyleSheet("background-color: #969696")

        elif THEME == 'dark':
            self.ui.title_frame.setStyleSheet("QFrame#title_frame {\n"
"    background-color: #898989;\n"
"}\n"
"\n"
"QPushButton {\n"
"    background-color: rgba(0,0,0,0);\n"
"    color: #484848;\n"
"    font: bold;\n"
"    font-size: 15px;\n"
"    font-family: entypo;\n"
"}\n"
"\n"
"QPushButton:hover {\n"
"    color: #E8960C;\n"
"}\n"
"\n"
"QPushButton:pressed {\n"
"    color: #FF8000;\n"
"    padding-left: 5px;\n"
"    padding-top: 5px;\n"
"}")
            self.ui.lineEdit.setStyleSheet("background-color: #8f8f8f;\n"
"border-radius: 5px;\n"
"color: #5c5c5c;\n"
"padding-left: 8px;")
            self.ui.tabWidget.setStyleSheet("QTabWidget::scroller {\n"
"    width: 0px;\n"
"}\n"
"\n"
"QTabWidget::pane {\n"
"    background: #96989f;\n"
"}\n"
"\n"
"QTabWidget::tab-bar:top {\n"
"    top: 1px;\n"
"}\n"
"\n"
"QTabWidget::tab-bar:bottom {\n"
"    bottom: 1px;\n"
"}\n"
"\n"
"QTabWidget::tab-bar:left {\n"
"    right: 1px;\n"
"}\n"
"\n"
"QTabWidget::tab-bar:right {\n"
"    left: 1px;\n"
"}\n"
"\n"
"QTabBar::tab:selected {\n"
"    background: #96989f;\n"
"    color: #000;\n"
"}\n"
"\n"
"QTabBar::tab:!selected {\n"
"    border-right: 1px solid #5d656d;\n"
"    border-left: 1px solid #5d656d;\n"
"    background: #8a8d97;\n"
"    color: #5a6068;\n"
"}\n"
"\n"
"QTabBar::tab:!selected:hover {\n"
"    background: #93959d;\n"
"    color: #5a6068;\n"
"}\n"
"\n"
"QTabBar::tab:top:!selected {\n"
"    margin-top: 1px;\n"
"}\n"
"\n"
"QTabBar::tab:bottom:!selected {\n"
"    margin-bottom: 1px;\n"
"}\n"
"\n"
"QTabBar::tab:top, QTabBar::tab:bottom {\n"
"    min-width: 24ex;\n"
"    margin-right: -1px;\n"
"    padding: 10px;\n"
"}\n"
"\n"
"QTabBar::tab:top:selected {\n"
"    border-bottom-color: none;\n"
"}\n"
"\n"
"QTabBar::tab:bottom:selected {\n"
"    border-top-color: none;\n"
"}\n"
"\n"
"QTabBar::tab:top:last, QTabBar::tab:bottom:last,\n"
"QTabBar::tab:top:only-one, QTabBar::tab:bottom:only-one {\n"
"    margin-right: 0;\n"
"}\n"
"\n"
"QTabBar::tab:left:!selected {\n"
"    margin-right: 3px;\n"
"}\n"
"\n"
"QTabBar::tab:right:!selected {\n"
"    margin-left: 3px;\n"
"}\n"
"\n"
"QTabBar::tab:left, QTabBar::tab:right {\n"
"    min-height: 8ex;\n"
"    margin-bottom: -1px;\n"
"    padding: 10px 5px 10px 5px;\n"
"}\n"
"\n"
"QTabBar::tab:left:selected {\n"
"    border-left-color: none;\n"
"}\n"
"\n"
"QTabBar::tab:right:selected {\n"
"    border-right-color: none;\n"
"}\n"
"\n"
"QTabBar::tab:left:last, QTabBar::tab:right:last,\n"
"QTabBar::tab:left:only-one, QTabBar::tab:right:only-one {\n"
"    margin-bottom: 0;\n"
"}\n"
"\n"
"QTabBar::close-button {\n"
"    background: rgba(0,0,0,0);\n"
"    image: url(:/img/img/cross.png);\n"
"    subcontrol-position: left;\n"
"    border-radius: 7px;\n"
"    padding: 2px;\n"
" }\n"
"\n"
"QTabBar::close-button:hover {\n"
"    background: #a6a6a9;\n"
"cursor: pointing-hand\n"
"}")
        self.ui.bottom_frame.setStyleSheet("background-color: rgb(45, 45, 45);")
        self.ui.content_frame.setStyleSheet("background-color: rgb(45, 45, 45)")